substrate/pkg/controller/substrate/cluster/addons/karpenter.go

```diff
@@ -0,0 +1,96 @@
+package addons
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/karpenter/pkg/apis"
+	"github.com/awslabs/kit/substrate/pkg/apis/v1alpha1"
+	"github.com/awslabs/kit/substrate/pkg/controller/substrate/cluster"
+	"github.com/awslabs/kit/substrate/pkg/utils/discovery"
+	"github.com/awslabs/kit/substrate/pkg/utils/helm"
+	"github.com/awslabs/kit/substrate/pkg/utils/kubectl"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type Karpenter struct {
+	EC2 *ec2.EC2
+}
+
+func init() {
+	runtime.Must(apis.AddToScheme(scheme.Scheme))
+}
+
+var provisioner = `
+apiVersion: karpenter.sh/v1alpha5
+kind: Provisioner
+metadata:
+  name: default
+spec:
```

Comment:
We need to provide ClusterDNS for kubeletConfiguration.
Also, don't we need to provide labels anymore? (I will check the latest changes in karpenter but just want to keep a track of this)

Reply:
This was just a sample. If we want to do the real provisioner config, we're gonna need to wait for the next karpenter release.